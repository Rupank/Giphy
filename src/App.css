* {
  box-sizing: border-box;
}

body {
  background-color: #1c1c1c;
  color: white;
}

.searchBox {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background: black;
  height: 10vh;
  margin-bottom: 20px;
}

.searchBox input {
  width: 60%;
  height: 60%;
  border-radius: 10em;
  border: none;
  outline: none;
  padding-left: 20px;
  color: black;
  font-size: medium;
}

.grid {
  width: 96%;
  columns: 6;
  column-gap: 10px;
  margin-left: 2%;
  margin-right: 2%;
}

img {
  object-fit: cover;
  padding: 5px;
  width: 100%;
  height: 100%;
}

@media screen and (max-width: 1200px) {
  .grid {
    columns: 5;
  }
}

@media screen and (max-width: 768px) {
  .grid {
    columns: 4;
  }
  .searchBox {
    height: 8vh;
  }
}

@media screen and (max-width: 480px) {
  .searchBox {
    height: 5vh;
  }

  .grid {
    column-gap: 5px;
    columns: 3;
  }
}
