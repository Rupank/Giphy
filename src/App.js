import React from 'react';
import './App.css';
import { Route, Switch } from 'react-router-dom';
import Search from './components/Search';
import SearchResults from './components/Results'

function App() {
  return (
    <>
      <Search />
      <Switch>
        <Route exact path='/' component={SearchResults} />
      </Switch>
    </>
  );
}

export default App;
