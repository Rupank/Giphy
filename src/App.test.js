import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from './store/createStore';

test('renders App Component', () => {

  const { asFragment } = render(<Provider store={store}>
    <BrowserRouter>
      <App />
    </BrowserRouter>
  </Provider>);
  expect(asFragment()).toMatchSnapshot();
});
