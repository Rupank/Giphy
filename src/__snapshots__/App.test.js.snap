// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App Component 1`] = `
<DocumentFragment>
  <div
    class="searchBox"
  >
    <input
      placeholder="Type a word and Hit Enter"
    />
  </div>
  <div
    class="grid"
  />
</DocumentFragment>
`;
